runtime/transformations/components/index-operations/reset_index_and_save_as_column_100_77258f6a-db49-42a6-8133-7fcfd5bffddb.py

```diff
@@ -0,0 +1,141 @@
+"""
+
+# add your own imports here, e.g.
+import pandas as pd
+
+from hdutils import (
+    ComponentInputValidationException,
+    parse_value,  # noqa: E402
+)
+
+# ***** DO NOT EDIT LINES BELOW *****
+# These lines may be overwritten if component details or inputs/outputs change.
+COMPONENT_INFO = {
+    "inputs": {
+        "df_to_reset_index": {"data_type": "DATAFRAME"},
+        "index_name": {"data_type": "STRING", "default_value": "timestamp"},
+    },
+    "outputs": {
+        "df_with_reset_index": {"data_type": "DATAFRAME"},
+    },
+    "name": "Reset Index and Save as Column",
+    "category": "Index Operations",
+    "description": "Uses the pandas reset_index method to reset the index of a DataFrame and save the previous index as a column.",  # noqa: E501
+    "version_tag": "1.0.0",
+    "id": "77258f6a-db49-42a6-8133-7fcfd5bffddb",
+    "revision_group_id": "175f432d-0efd-499c-bf62-85aa42761de9",
+    "state": "RELEASED",
+    "released_timestamp": "2024-12-03T15:49:27.058488+00:00",
+}
+
+from hdutils import parse_default_value  # noqa: E402, F401
+
+
+def main(*, df_to_reset_index, index_name="timestamp"):
+    # entrypoint function for this component
+    # ***** DO NOT EDIT LINES ABOVE *****
+
+    # check whether index_column is in the columns and is appropriate as the index of a dataframe
+    if index_name in df_to_reset_index.columns:
+        raise ComponentInputValidationException(
+            f"There is already a column named {index_name} in the DataFrame passed to df_to_reset_index.",
+            invalid_component_inputs=["df_to_reset_index", "index_name"],
+        )
+
+    reset_df = df_to_reset_index.reset_index(drop=False, names=index_name, allow_duplicates=False)
+    return {"df_with_reset_index": reset_df}
+
+
+# Testing
+try:
+    import pytest
+except ImportError:
+    pass
+else:
+
+    @pytest.fixture()
+    def timeseries_dataframe():
+        values = [1.0, 1.2, 1.2]
+        timestamps = pd.to_datetime(
+            [
+                "2019-08-01T15:45:36.000Z",
+                "2019-08-02T11:33:41.000Z",
+                "2019-08-03T11:57:41.000Z",
+            ],
+            format="%Y-%m-%dT%H:%M:%S.%fZ",
+        ).tz_localize("UTC")
+
+        ts_df = pd.DataFrame({"timestamp": timestamps, "value": values})
+
+        return ts_df
+
+    def test_run_from_test_wiring(timeseries_dataframe):
+        output_df = main(
+            **{
+                inp_wiring["workflow_input_name"]: parse_value(
+                    inp_wiring["filters"]["value"],
+                    COMPONENT_INFO["inputs"][inp_wiring["workflow_input_name"]]["data_type"],
+                    nullable=True,
+                )
+                for inp_wiring in TEST_WIRING_FROM_PY_FILE_IMPORT["input_wirings"]
+                if inp_wiring.get("adapter_id", "direct_provisioning") == "direct_provisioning"
+            }
+        )["df_with_reset_index"]
+
+        pd.testing.assert_frame_equal(output_df, timeseries_dataframe)
```

Comment:
Du prüfst mit diesem Test zwei Sachen ab: 
1) passt das wiring zur Funktion
2) liefert es die Ergebnisse, die es liefern sollte. 

Dies entspricht nicht dem best practices https://medium.com/@kaanfurkanc/unit-testing-best-practices-3a8b0ddd88b5, dass man nur ein concern testen soll.

Da sich TEST_WIRING_FROM_PY_FILE_IMPORT ändert, wenn man im Frontend, mal etwas anderes einstellt, und dann hdctl sync pull macht, und dementsprechend die Gefahr gegeben ist, dass 1) zutrifft aber 2) nicht, würde ich diesen Test auf 2 getrennte Tests aufsplitten. Einer, in dem du prüfst, ob kein Fehler auftaucht, wenn man das Wiring verwende, (z.B. in dem du prüfst, ob im zurückgegeben Dict ein nicht leerer Eintrag unter "df_with_reset_index" auftaucht und einer, wo du einen definierten DF reingibst, und einen definierten DF zurückkriegst.